jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -148,9 +153,12 @@ object XGBoost extends Serializable {
    * @param missing the value represented the missing value in the dataset
    * @param featureCol the name of input column, "features" as default value
    * @param labelCol the name of output column, "label" as default value
+   * @param baseMarginCol the name of the column with the initial prediction,
+   *                      "baseMargin as default value
    * @throws ml.dmlc.xgboost4j.java.XGBoostError when the model training is failed
    * @return XGBoostModel when successful training
    */
+  // scalastyle:off
```

Comment:
would you please do it in this PR? 

about how to make API easier to discover, I do not think it's a problem...think about all supported functionalities in XGBoost, most of them are enabled/disabled through `params`

Actually, except ones like `numWorkers` and `round` which are directly related to the training process, obj/evalTrait/missing/etc. should go to `params`